src/app/reminder-list/page.tsx

```diff
@@ -1,86 +1,207 @@
+      patientName: "Alian Haidar",
+      repeatPeriod: RepeatPeriod.Monthly,
+      startDate: "1999-09-09",
+      endDate: "2000-01-01",
+      timeOfDay: "10:45PM",
+      messageToPatient: "this is a hard-coded message for reminder #2"
+    }
+  ];
+
+  const [editing, setEditing] = useState<boolean>(false);
+
+  const handleReminderCardClick = (i: number) => {
+    setEditing(true);
+  };
+
+  const handleAddNewButton = () => {
+    setEditing(true);
+  };
 
-export default function Home() {
-  const [showForm, setShowForm] = useState(false);
-  const [showAddButton, setshowAddButton] = useState(false);
-  const [showHeader, setshowHeader] = useState("");
+  const handleSaveClick = () => {
+    setEditing(false);
+  };
 
-  const handleAddButtonClick = () => {
-    setShowForm(true);
-    setshowAddButton(true);
-    setshowHeader("Add Reminder");
+  const handleCancelClick = () => {
+    setEditing(false);
   };
-  const handleBackButtonClick = () => {
-    setShowForm(false);
-    setshowAddButton(false);
-    setshowHeader("");
+
+  const handleEdit = (
+    event: ChangeEvent<HTMLInputElement>,
+    newReminder: Reminder
+  ) => {
+    const index = reminders.findIndex(
+      (item) => item.reminderID === newReminder.reminderID
+    );
+    if (index === -1) {
+      reminders.push(newReminder);
+    } else {
+      reminders[index] = newReminder;
+    }
   };
 
   return (
     <>
-      <div className="flex h-screen flex-col">
-        {/* main */}
-        <Header
-          title={showHeader}
-          type="primary"
-          iconLeft={
-            <IconButton aria-label="back" className="bg-base-100">
-              <ChevronLeftIcon
-                fontSize="large"
-                sx={{ padding: 1 }}
-                onClick={handleBackButtonClick}
-              />
-            </IconButton>
-          }
-          iconRight={
-            showAddButton ? (
-              <Button>Add</Button>
-            ) : (
+      {editing ? (
+        <div className="flex h-screen flex-col">
+          <Header
+            title="Edit Reminder"
+            type="primary"
+            iconLeft={
               <IconButton aria-label="back" className="bg-base-100">
-                <AddIcon
-                  fontSize="large"
-                  onClick={handleAddButtonClick}
-                  sx={{ padding: 1 }}
-                />{" "}
+                <LuStepBack
+                  className="text-primary"
+                  size={30}
+                  strokeWidth={1.3}
+                  onClick={handleCancelClick}
+                />
               </IconButton>
-            )
-          }
-        />
-
-        <div className="w-full flex-1 overflow-y-auto bg-base-100">
-          {showForm ? (
-            <div className="  flex flex-col gap-4 px-6 py-2">
-              <AddReminer />
+            }
+            iconRight={
+              <IconButton aria-label="back" className="bg-base-100">
```

Comment:
Do not wrap the icon with `<IconButton>`. `<Header>` already wraps incoming icon with a `<button>`.